src/labels.js

```diff
@@ -69,10 +49,18 @@ class Labels {
     }
   }
 
-  static initMetadata (wallet) {
-    return Metadata.fromMasterHDNode(wallet._metadataHDNode, METADATA_TYPE_LABELS);
+  get readOnly () {
+    return false;
+  }
+
+  syncWallet () {
+    const syncWallet = () => {
+      this._syncWallet(() => Promise.resolve(), (e) => Promise.reject(e));
```

Comment:
I am not sure why labels class is needed but I have not check carefully the code. I would say that the track of the last index is something that semantically belongs to hdaccount...

About the syncWallet. I don't really like this model where we do stuff and syncWallet is called magically, but the problem is not specific from labels.. it is more a general architecture problem.. .so...